File: fadl_pos/fadl_pos/report/branch_item_summary/branch_item_summary.py
# Copyright (c) 2026, FadlTech team and contributors

from __future__ import annotations



import frappe
from frappe import _


def execute(filters=None):
	columns, data = get_columns(), get_data(filters)
	return columns, data


def get_data(filters):
	query = """
        SELECT
            sed.item_code,
            sed.item_name,
            SUM(sed.qty) AS qty
        FROM `tabStock Entry` se
        INNER JOIN `tabStock Entry Detail` sed ON se.name = sed.parent
        WHERE
            se.docstatus = 1
            AND se.company = %(company)s
            AND se.posting_date >= %(from_date)s AND se.posting_date <= %(to_date)s
            AND sed.t_warehouse = %(warehouse)s
        GROUP BY sed.item_code, sed.t_warehouse
    """

	return frappe.db.sql(
		query,
		{
			"company": filters.get("company"),
			"from_date": filters.get("from_date"),
			"to_date": filters.get("to_date"),
			"warehouse": filters.get("warehouse"),
		},
		as_dict=True,
	)


def get_columns():
	return [
		{
			"label": _("Item Code"),
			"fieldname": "item_code",
			"fieldtype": "Link",
			"options": "Item",
			"width": 150,
		},
		{
			"label": _("Item Name"),
			"fieldname": "item_name",
			"fieldtype": "Data",
			"width": 200,
		},
		{"label": _("Quantity"), "fieldname": "qty", "fieldtype": "Float", "width": 100},
	]

File: fadl_pos/fadl_pos/report/branch_item_summary/branch_item_summary.js
// Copyright (c) 2025, Ali Raza and contributors
// For license information, please see license.txt

frappe.query_reports["Branch Item Summary"] = {
	filters: [
		{
			fieldname: "company",
			label: __("Company"),
			fieldtype: "Link",
			options: "Company",
			reqd: 1,
			default: frappe.defaults.get_default("company"),
			on_change: function () {
				frappe.query_report.set_filter_value("warehouse", null);
			},
		},
		{
			fieldname: "warehouse",
			label: __("Warehouse"),
			fieldtype: "Link",
			options: "Warehouse",
			reqd: 1,
			get_query: () => {
				return {
					filters: {
						company: frappe.query_report.get_filter_value("company"),
						is_group: 0,
					},
				};
			},
		},
		{
			fieldname: "from_date",
			label: __("From Date"),
			fieldtype: "Date",
			reqd: 1,
			default: frappe.datetime.get_today(),
		},
		{
			fieldname: "to_date",
			label: __("To Date"),
			fieldtype: "Date",
			reqd: 1,
			default: frappe.datetime.get_today(),
		},
	],
};
